File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

 // Time complexity O(n)
 // Space complexity OLog(n)
func inorderTraversal(root *TreeNode) []int {
    data := make([]int, 0, 10)
	inorder(root, &data)
	return data
}

func inorder(root *TreeNode, data *[]int) {
	if root == nil {
		return
	}

	inorder(root.Left, data)
	*data = append(*data, root.Val)
	inorder(root.Right, data)
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        var response = new ArrayList<Integer>();
        inorder(root, response);
        return response;
    }
    
    public void inorder(TreeNode root, ArrayList<Integer> response) {
        if(root == null)
            return;
        
        inorder(root.left, response);
        response.add(root.val);
        inorder(root.right, response);
        
    }
}